docker_mcp/core/compose_manager.py

```
@@ -337,16 +338,26 @@ async def write_compose_file(self, host_id: str, stack_name: str, compose_conten
         Returns:
             Full path to the written compose file
         """
-        compose_base_dir = await self.get_compose_path(host_id)
+        try:
+            async with asyncio.timeout(15.0):
+                compose_base_dir = await self.get_compose_path(host_id)
+        except TimeoutError:
+            logger.error("Get compose path timed out", host_id=host_id)
```

Ruff (0.14.4)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)